symforce/opt/optimizer.tcc

```diff
@@ -65,10 +32,11 @@ Optimizer<ScalarType, NonlinearSolverType>::Optimizer(const optimizer_params_t&
 template <typename ScalarType, typename NonlinearSolverType>
 template <typename... NonlinearSolverArgs>
 Optimizer<ScalarType, NonlinearSolverType>::Optimizer(
-    const optimizer_params_t& params, std::vector<Factor<Scalar>>&& factors, const Scalar epsilon,
-    const std::string& name, std::vector<Key>&& keys, bool debug_stats, bool check_derivatives,
+    const optimizer_params_t& params, std::vector<Factor<Scalar>> factors, const Scalar epsilon,
+    std::string name, std::vector<Key> keys, bool debug_stats, bool check_derivatives,
     NonlinearSolverArgs&&... nonlinear_solver_args)
     : factors_(std::move(factors)),
+      name_(std::move(name)),
```

Comment:
Did this one just throw away the name? :grimacing:

Reply:
my bad, didn't realize name was also used later. I fixed it.

Reply:
It also feels weird that both optimizer and nonlinear solver need a name.

Reply:
Yeah the `name_` in all of these classes is primarily (only?) for logging and TicToc purposes, given the places we need to access it it's reasonably convenient to just make it an instance variable of the classes that use it, but we could pass it around instead